doc/ImplementingApplications/Steps2IntegrateElasticsearch/Steps2IntegrateElasticsearch.md

```diff
@@ -0,0 +1,255 @@
+## Integrate Elasticsearch (ES)
+
+Disclaimer: This guide is not meant to be a manual for ES itself,
+it only covers areas implemented in ApplicationPattern and guides how to use
+the class ElasticsearchService in your application. For more information about
+Elasticsearch, go to [official documentation](https://www.elastic.co/).
+
+### What is ES and why should my application use it?
+
+ES is a distributed, free and open search and analytics engine for all types of data, including textual, numerical, geospatial, structured, and unstructured. Your application should consider using it, if:
+- it will store large amounts of data
+- it will store data often
+
+### Brief introduction to ES buzzwords
+
+Basic building block in ES is `document`. By this word, ES understands one item of `data`. 
+Documents have own `_id` and `_source`, where the actual data is stored. Documents are grouped
```

Comment:
Also, i think it would be better if we add bit more information about each term mentioned, so that a new person who is trying to include ES in his application would easily understand the concepts